base/reduce.jl

```diff
@@ -453,16 +453,23 @@ prod(a) = mapreduce(identity, mul_prod, a)
 
 ## maximum & minimum
 
+# these propagate NaN correctly only in the second argument
+_max_fast(x, y) = ifelse(x > y, x, y)
```

Comment:
Could perhaps just define

```jl
_op_fast(::typeof(max), x, y) = ifelse(x > y, x, y)
_op_fast(::typeof(min), x, y) = ifelse(x < y, x, y)
```

and just use _`_op_fast(op, v, f(ai)` in `mapreduce_impl`?

Reply:
I like this idea.